test/headers.js

```diff
@@ -0,0 +1,240 @@
+import createTestServer from 'create-test-server';
+import FormData from 'form-data';
+import test from 'ava';
+import ky from '..';
+
+const echoHeaders = (request, response) => {
+	request.resume();
+	response.end(JSON.stringify(request.headers));
+};
+
+test('`user-agent`', async t => {
+	const server = await createTestServer();
+	server.get('/', echoHeaders);
+
+	const headers = await ky.get(server.url).json();
+	t.is(
+		headers['user-agent'],
+		'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)'
+	);
+});
+
+test('`accept-encoding`', async t => {
+	const server = await createTestServer();
+	server.get('/', echoHeaders);
+
+	const headers = await ky.get(server.url).json();
+
+	t.is(headers['accept-encoding'], 'gzip,deflate');
+});
+
+test('does not override provided `accept-encoding`', async t => {
+	const server = await createTestServer();
+	server.get('/', echoHeaders);
+
+	const headers = await ky
+		.get(server.url, {
+			headers: {
+				'accept-encoding': 'gzip'
+			}
+		})
+		.json();
+	t.is(headers['accept-encoding'], 'gzip');
+});
+
+test('does not remove user headers from `url` object argument', async t => {
+	const server = await createTestServer();
+	server.get('/', echoHeaders);
+
+	const headers = await ky
+		.get(server.url, {
+			hostname: server.hostname,
+			port: server.port,
+			responseType: 'json',
+			protocol: 'http:',
```

Comment:
Most of these properties are not supported by Ky. I guess we could keep it anyway, if having identical tests is worth some cruft, which I'm not entirely against. But it probably makes the test less valuable, which isn't entirely obvious just from glancing at it.

I suppose what I'm getting at is that we should decide how much we want to tailor these tests for Ky vs leave them alone.

Reply:
I think we should tailor the tests for Ky. Having unsupported properties in there just makes it confusing to read the tests.

@braedongough Can you remove unsupported properties?